airflow/utils/net.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from airflow.configuration import (conf, AirflowConfigException)
+import socket
+
+
+def get_hostname():
+    """
+    Fetch the hostname from the config or using `socket.getfqdn` as a fallback.
+    """
+    try:
+        return str(conf.get('core', 'hostname'))
```

Comment:
This supports overriding `getfqdn` with a fixed string, but does not support overriding with another method (i.e. `gethostname`, `gethostbyname`, or something a bit more customized). It would be ideal to not have to hard-code the value but allow user to implement `get_hostname`, especially when running Airflow in a distributed setup where it would be useful to identify which machine the task ran on.

One potential option is passing in a module in airflow.cfg instead of a string, something roughly like the following:
```
[core]
net_module = airflow.utils.net
```

And create a base implementation:
```
def get_hostname():
    return socket.getfqdn()
```

It can be used like:
```
net_module = import_module(conf.get('core', 'net_module'))
get_hostname = getattr(net_module, 'get_hostname')
hostname = get_hostname()
```

This will allow user to override the module and the get_hostname method.

Reply:
I like the idea. Let me update this PR.